system-addon/content-src/components/CollapsibleSection/CollapsibleSection.jsx

```diff
@@ -137,6 +141,16 @@ class CollapsibleSection extends React.PureComponent {
   componentWillMount() {
     this.props.document.addEventListener(VISIBILITY_CHANGE_EVENT, this.enableOrDisableAnimation);
   }
+  componentWillUpdate(nextProps) {
+    // Check if we're about to go from expanded to collapsed
+    if (!getCollapsed(this.props) && getCollapsed(nextProps)) {
+      // This next line forces a layout flush of the section body, which has a
+      // max-height style set, so that the upcoming collapse animation can
+      // animate from that height to the collapsed height. Without this, the
+      // update is coalesced and there's no animation from no-max-height to 0.
```

Comment:
Not sure if I want to know how you figured this out 😨

Reply:
It was mostly accidental… I was doing `console.log(scrollHeight)` from `componentDidUpdate` to see how it was changing because at some point I had `setState` calls from there because the calculated dom height only could be gotten after it's been rendered. Things seemed to be fixed, so I started deleting the `console.log` statements… and things broke! o.O After more debugging and undoing, turns out restoring the `console.log(scrollHeight)` fixed things again…